src/nvim/os/lang.c

```diff
@@ -340,7 +342,7 @@ char *get_locales(expand_T *xp, int idx)
 void lang_init(void)
 {
 #ifdef __APPLE__
-  if (os_getenv("LANG") == NULL) {
+  if (!os_env_exists("LANG", true)) {
```

Comment:
it was only checking NULL, which means the arg should be `false` ?
```suggestion
  if (!os_env_exists("LANG", false)) {
```

Reply:
`os_getenv` returns `NULL` for empty string